### ISSUE TITLE
v3.4-alpha Feedback

### ISSUE THREAD

**Issue by nmsmith22389** [nmsmith22389/HotLoot#26]
# <span style="color: #ffbb00;">Test The New Version!</span>
There is a new version of **HotLoot** in development that completely changes the way HL works!

This new version *(v3.4-alpha)* gets rid of the loot filters and replaces them with **custom filters**. Custom filters allow you to define exactly what you want looted and don't rely on predefined options. You can have as many filters as you want and each filter can have many conditions that need to be met for an item to be looted. It sort of works the same way conditions in WeakAuras work.

<span style="color: #ff0000;">This is an **ALPHA** version and many things still do not work. **USE AT YOUR OWN RISK!**</span>

To try the alpha version either download the file manually from the files section or right click on the HotLoot addon in the twitch/curse app and select to receive alpha releases. Please back up your setting to be extra safe as this version changes/removes many settings.

The whole reason I'm releasing this as a early alpha test is because I require the feedback of HotLoot's many dedicated users.

### Thanks!

~ Neil

**Comment by dsblack115** [comment]
I tried this out and it seemed to work except for the skinning mode.  Normally it deletes leftover items/junk which is really useful if you do not want all of the clutter in your bags, but it was looting everything.  The main filters I set up were for trade goods like Cloth, Leather, and Herbs.

**Comment by nmsmith22389** [comment]
I'm confused as to what the problem is. So you have the `Skinning Mode` option on, filters are set up, and it just loots everything but doesn't actually delete anything?

**Comment by dsblack115** [comment]
Yes, basically whenever I have Skinning Mode on it will delete leftover items/junk, and I see the notification in the chat window that it deleted leftover items and whatever the items are.  With the updated Alpha version, it loots everything and does not delete like it normally does, and no chat notification for deletion.

**Comment by nmsmith22389** [comment]
In the newest iteration of the alpha **(which I can upload soon)** I've completely rebuilt the skinning mode deleting process. Hopefully after all this new work I've done it will work well.

In the meantime I'm dying to hear what you think about the rest of the alpha version.

**Comment by dsblack115** [comment]
Conditions don't need to be in any particular order do they?
![capture](https://user-images.githubusercontent.com/22847171/32401338-9f242846-c0e2-11e7-9040-86692cfd9ccf.PNG)

**Comment by dsblack115** [comment]
It is my understanding from the above setup for custom filter 'Junk' that it should only loot poor items greater than 5g in value.  It does not seem to be working that way as it has looted junk less than 5g in value.

**Comment by nmsmith22389** [comment]
I'll look into it... It all seems ok but I'll see tonight if there's a bug somewhere in the value settings.

**Comment by Yakehira** [comment]
Add the previous available filters as pre-defined custom filters.

Use LOOT_READY, not LOOT_OPENED to generate your loot table in core.lua

LOOT_OPENED will fire when the Loot frame opens, LOOT_READY fires when the loot is generated so the info you take form LOOT_OPENED is already fully available from LOOT_READY and you can loot the moment LOOT_READY fires.

Id also recommend you to consider re-parenting the LootFrame to a Hidden parent when CloseLoot option is enabled, and set its strata and parent back after LOOT_CLOSED event. This will remove all the glitchyness that comes from enabling close loot option, the constant lootframe/tooltips that popup for a microsecond are gone with this method and it works flawlessly, you might also want to keep the lootframe open if the inventory is full or as an option.

**Comment by nmsmith22389** [comment]
@Yaiklai 
Those are some great suggestions! Thanks!

- Predefined filters are a great idea and I'll be sure to implement them in the next alpha patch.
- I've looked into `LOOT_READY` as an alternative but I've never seen a case of the loot info not being available on `LOOT_OPENED` so I'm still not seeing any benefit to switching... Maybe if you could elaborate more.
- Yes, getting rid of the loot frame when `Auto-close Loot Window` is enabled might be a better way. I wonder if hooking/replacing the function might even work better than reparenting it.

**Comment by nmsmith22389** [comment]
@dsblack115 
I'm not sure what quite happened there but I've since made updates and its working perfectly for me. I'll be releasing a new alpha patch soon so try that one when it comes out and let me know how it goes.

**Comment by Yakehira** [comment]
There are rare cases where the game lets you open the loot frame out of range, but can not loot. LOOT_OPENED will fire, but LOOT_READY will not in such cases. Anyway its not that big of a deal, but LOOT_READY is the event you should use if you want it to truly only work when loot is available and be as fast as it can, as it fires before LOOT_OPENED :)

**Comment by nmsmith22389** [comment]
@Yaiklai 
Well that shouldn't be too hard to change. I do also have a problem with HotLoot thinking the items have been looted when you get a *"Loot Busy"* error and showing them in the loot monitor... I've tried several workarounds but have yet to fix it... maybe if you have an idea.

**Comment by Yakehira** [comment]
CHAT_MSG_LOOT/CHAT_MSG_MONEY, that only fires when you looted it, but you will need to make sure its yours and not a party member and you can make it so it only listens to it after LOOT_READY/OPENED or LOOT_SLOT_CLEARED so you don't listen to it when you are not looting it yourself unless you want to display assigned loot from master looter but I don't think it should track that. Only using LOOT_SLOT_CLEARED might also work but less perfect as it fires regardless who clears the frame and you only get the slot, not the receiver.

**Comment by nmsmith22389** [comment]
Hey guys! Sorry for the lull in the releases. 😢 My MacBook Pro fried it's logic board and they want freaking $600 to replace it. So in the meantime I'm working on a shitty Windows laptop and its terrible. **But**, the good news is I have a new alpha release that I think is about ready to go into beta. I'll throw it up today or tomorrow. I added a lot of the stuff people suggested!

**Comment by dsblack115** [comment]
Hey Neil, no worries.  I have not been playing for a while, I burn myself out quick playing this game.  Still having the original issue with the loot value filter previously mentioned in this thread.  Not a big deal, since there are no errors and the addon in general is working smoothly.

**Comment by nmsmith22389** [comment]
Yeah sorry! I haven't had much time to play either. I was contracted to work on a big programming project. I'll get this fixed as soon as I can and try and get this new version up on curse.
